projects/dotcms-field-elements/src/components/dot-date/dot-date.tsx

```diff
@@ -88,27 +100,24 @@ export class DotDateComponent {
     @Listen('_valueChange')
     emitValueChange(event: CustomEvent) {
         event.stopImmediatePropagation();
-        this.valueChange.emit(event.detail);
+        const valueEvent: DotFieldValueEvent = event.detail;
+        this.value = valueEvent.value;
+        this.valueChange.emit(valueEvent);
     }
 
     @Listen('_statusChange')
     emitStatusChange(event: CustomEvent) {
         event.stopImmediatePropagation();
-        const statusEvent: DotFieldStatusEvent = event.detail;
+        const statusEvent: DotInputCalendarStatusEvent = event.detail;
```

Comment:
`statusEvent` could get confused with the event itself this is only caching the value of the event.